navapbc/wic-mt-demo-project-eligibility-screener
# Configure chromatic to support UAT

## Ticket

https://wicmtdp.atlassian.net/browse/WMDP-39

## Changes
> What was added, updated, or removed in this PR.

- Install chromatic as a dev dependency
- Configure chromatic as a CI job
- Configure chromatic to be run during local development

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

For this project, we are using [Chromatic](https://chromatic.com) to host our storybook. I've configured a Github Actions CI job to automatically push a storybook build to chromatic whenever a PR has been merged to main and created a Github Secret with the chromatic project token.

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

1. Test the chromatic local development flow: Follow the new instructions in the `app/README.md`
2. Test the chromatic CI pipeline flow: Review the log for the test CI pipeline: https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/runs/7555310017?check_suite_focus=true

To access Chromatic storybook builds:

1. Log into Chromatic
2. Click on the project and then click on "Builds" in the left sidebar
3. Select the build you are interested in. Look at the branch name. For UAT, make sure to select a build for the `main` branch.
4. Click the "View storybook" button on the right
